CMakeLists.txt

```diff
@@ -72,6 +73,15 @@ set(CMAKE_CXX_STANDARD 14)  # specify the C++ standard
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
+# coverage
+if (ENABLE_TESTING AND ENABLE_COVERAGE)
+    add_compile_options(--coverage)
+    add_compile_options(-g)
+    add_compile_options(-O0)
+
+    set(COVERAGES --coverage)
```

Comment:
Is this option valid for clang ? If not, please check the cmake compiler variable.

Reply:
Yes, valid for gcc/clang.